### DISCUSSION TITLE
disable "-Wcovered-switch-default"

### DISCUSSION THREAD

**Discussion by frokaikan** [nlohmann/json#3539] [Ideas]
Maybe add 
```c++
#pragma clang diagnostic ignored "-Wcovered-switch-default"
```
nearby line 2370 in `single_include/nlohmann/json.hpp`?
```
// disable documentation warnings on clang
#if defined(__clang__)
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wdocumentation"
    #pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
    #pragma clang diagnostic ignored "-Wcovered-switch-default" // maybe add near here?
#endif
```
Then compile with `-Wcovered-switch-default` and clang reports nothing.

**Comment by falbrechtskirchinger** [comment]
Ideally, we should just fix the source of the warning. I'll have a look.

**Reply by falbrechtskirchinger** [reply]
Refactoring all switch statements sounds like a good idea if anyone wants to do that.

Remove `default` where appropriate and explicitly check inputs against all enumerators in an assert or something to that effect.

**Reply by nlohmann** [reply]
I guess then there is another warning about a missing default case. This sounds familiar.

**Reply by gregmarr** [reply]
Yes, there are two different related warnings.  It is impossible to write code that covers both of them at the same time.  One complains if a switch doesn't have a default, and the other complains if a switch has a default but exhaustively covers an enum's values.

**Reply by gregmarr** [reply]
https://github.com/nlohmann/json/pull/2927#issuecomment-897981299 
https://github.com/nlohmann/json/pull/555#issuecomment-294304901
#889
#890

**Reply by nlohmann** [reply]
Thanks for looking up the issues. I guess we need to leave it as is.
